src/pilot/dell_nfv.py

```diff
@@ -0,0 +1,376 @@
+            OC_AUTH_URL, OC_PROJECT_ID, OC_USERNAME, OC_PASSWORD = \
+                CredentialHelper.get_overcloud_creds()
+            keystone_auth = v2.Password(username=OC_USERNAME,
+                                        password=OC_PASSWORD,
+                                        tenant_name=OC_PROJECT_ID,
+                                        auth_url=OC_AUTH_URL)
+            keystone_session = session.Session(auth=keystone_auth)
+            kwargs = {'auth_url': OC_AUTH_URL,
+                      'session': keystone_session,
+                      'auth': keystone_auth,
+                      'service_type': 'orchestration'}
+            ht_client = heat_client.Client('1', **kwargs)
+            try:
+                file_path = home_dir \
+                    + '/pilot/templates/overcloud/puppet/'\
+                    "services/dellnfv/createaggregate.yaml"
+                template = open(file_path)
+            except Exception as error:
+                message = "Exception {}: {}".format(
+                    type(error).__name__, str(error))
+                logger.error("{}".format(message))
+                raise Exception("The createaggregate.yaml file does not exist")
+            try:
+                file_path = home_dir \
+                    + '/pilot/templates/create_aggregate_environment.yaml'
+                environment = open(file_path)
+            except Exception as error:
+                message = "Exception {}: {}".format(
+                    type(error).__name__, str(error))
+                logger.error("{}".format(message))
+                raise Exception("The create_aggregate_environment.yaml"
+                                "file does not exist")
+            try:
+                stack = ht_client.stacks.create(stack_name=stack_name,
+                                                template=template.read(),
+                                                environment=environment.read())
+            except heatclient.exc.HTTPConflict as e:
+                error_state = e.error
+                raise Exception("Stack already exists : ",
+                                error_state, stack_name)
+            template.close()
+            environment.close()
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            raise Exception("Aggregate could not be"
+                            " created due to {}".format(message))
+
+    def post_deployment_tasks(self):
+        try:
+            # Reboot all the Dell NFV compute nodes
+            self.reboot_compute_nodes()
+
+            logger.info("Initiating post deployment tasks")
+
+            # create aggregate
+            self.create_aggregate()
+
+        except Exception as error:
+            message = "Exception {}: {}".format(
+                type(error).__name__, str(error))
+            raise Exception("{}".format(message))
```

Comment:
The above line could more clearly be written as "raise Exception(message)".  There really is no need for the format() call.

Reply:
Fixed